src/main/java/games/strategy/triplea/ai/AIUtils.java

```diff
@@ -131,7 +131,8 @@ public static int getIndexOfLastUnitMatching(final List<Unit> units, final Match
   }
 
   static List<Unit> interleaveCarriersAndPlanes(final List<Unit> units, final int planesThatDontNeedToLand) {
-    if (!(Match.anyMatch(units, Matches.unitIsCarrier()) && Match.anyMatch(units, Matches.unitCanLandOnCarrier()))) {
+    if (units.stream().noneMatch(Matches.unitIsCarrier())
+        || units.stream().noneMatch(Matches.unitCanLandOnCarrier())) {
```

Comment:
Just verifying for my own benefit that this expression still evaluates to `true` when `units` is empty.